crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -302,8 +302,14 @@ impl Decryptor for (matrix_sdk_base::crypto::OlmMachine, ruma::OwnedRoomId) {
         let (olm_machine, room_id) = self;
         let decryption_settings =
             DecryptionSettings { sender_device_trust_requirement: TrustRequirement::Untrusted };
-        let event =
-            olm_machine.decrypt_room_event(raw.cast_ref(), room_id, &decryption_settings).await?;
-        Ok(event.into())
+        match olm_machine
+            .try_decrypt_room_event(raw.cast_ref(), room_id, &decryption_settings)
+            .await?
+        {
+            RoomEventDecryptionResult::Decrypted(decrypted) => Ok(decrypted.into()),
+            RoomEventDecryptionResult::UnableToDecrypt(utd_info) => {
+                Ok(TimelineEvent::new_utd_event(raw.clone(), utd_info))
+            }
+        }
```

Comment:
Why is this change needed, out of curiosity? This is for the test timeline; is it operating that differently from the non-test timeline?

Reply:
It's needed to make the regression test ~"fails correctly". 
If not the test never detects the behavior change because it is using the mock `Decryptor` that continues to fail with an error when there is an UTD instead of return a OK UTDEvent type.

The underlying problem is that we use a big catch everything Error for all possible errors of the SDK. And here the change that created the regression should have resulted in a change of signature of the trait. It is not anymore Result<, ALLERRORS> but a Result <,SubsetOfErrors>?

Reply:
Alright, I'm slightly scared that we might be reproducing an error only in testing conditions, but this code is likely going away at some point…